numpy/distutils/misc_util.py

```diff
@@ -2273,13 +2267,13 @@ def generate_config_py(target):
     from numpy.distutils.system_info import system_info
     from distutils.dir_util import mkpath
     mkpath(os.path.dirname(target))
-    f = open(target, 'w')
-    f.write('# This file is generated by numpy\'s %s\n' % (os.path.basename(sys.argv[0])))
-    f.write('# It contains system_info results at the time of building this package.\n')
-    f.write('__all__ = ["get_info","show"]\n\n')
+    with open(target, 'w') as f:
+        f.write('# This file is generated by numpy\'s %s\n' % (os.path.basename(sys.argv[0])))
+        f.write('# It contains system_info results at the time of building this package.\n')
+        f.write('__all__ = ["get_info","show"]\n\n')
 
-    # For gfortran+msvc combination, extra shared libraries may exist
-    f.write("""
+        # For gfortran+msvc combination, extra shared libraries may exist
+        f.write("""
```

Comment:
Pretty funky indentation after this :). Could use a cleanup using dedent in another PR.